viewing_party/party.py

```diff
@@ -0,0 +1,192 @@
+    to user's watchlist, a list of dictionaries representing the movies the user wants to watch.
+    Returns updated user_data.
+    """
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    """
+    Checks if a movie is in the user's watchlist. If so, adds the movie to the 
+    list of watched movies, and removes the movie from watchlist.
+    Returns updated user_data.
+    """
+    user_watchlist = user_data["watchlist"]
+    user_watched = user_data["watched"]
+
+    for movie_entry in user_watchlist:
+        if movie_entry["title"] == title:
+            user_watched.append(movie_entry)
+            user_watchlist.remove(movie_entry)
+    return user_data
+
+# ——— WAVE 02 ———
+def get_watched_avg_rating(user_data):
+    """Calculates and returns the average rating of the movies in user's list of watched movies."""
+    ratings = 0
+    user_watched = user_data["watched"]
+
+    if user_watched == []:
+        return float(0)
+
+    else:
+        for movie_entry in user_watched:
+            ratings += movie_entry["rating"] 
+
+    return(ratings / len(user_watched))
+
+def get_most_watched_genre(user_data):
+    """
+    Builds a frequency dict to determine a user's most watched genre according to their
+    watched list, and returns the value as a string. Does not account for ties.
+    """
+    genre_frequencies = {}
+    user_watched = user_data["watched"]
+    
+    # Counts the occurrences of each genre in the user's watched list in a dictionary, where the dict keys are the genres
+    for movie_entry in user_watched:
+        if movie_entry["genre"] not in genre_frequencies:
+            genre_frequencies[movie_entry["genre"]] = 1
+        else:
+            genre_frequencies[movie_entry["genre"]] += 1
+
+    # Iterate through genre frequency dictionary to determine largest value. Uses a counter to compare genre frequencies
+    current_most_watched_genre = 0
+    for genre, frequency in genre_frequencies.items():
+        if frequency > current_most_watched_genre:
+            current_most_watched_genre = frequency
+            
+    for genre, frequency in genre_frequencies.items(): # this could be refactored to not use two loops… 🤠
+        if frequency == current_most_watched_genre:
+            return genre
+
+# ——— WAVE 03 ———
+def get_unique_watched(user_data):
+    """
+    Returns a list of dictionaries that represents a list of movies that are in the user's watched list, 
+    but are not in any of the user's friends watched lists. The movies are unique to the user.
+    """
+    not_unique_titles = []
+    user_watched = user_data["watched"] 
+    friends_watched = user_data["friends"]
+
+    for movie in range(len(user_watched)):
+        for x in range(len(friends_watched)): 
+            for item in friends_watched[x]["watched"]:
+                if user_watched[movie]["title"] in item.values():
```

Comment:
You could rewrite this a little more succinctly  as

```suggestion
                if user_watched[movie]["title"] == item["title"]
```